pkg/code-generator/collector/collector.go

```diff
@@ -0,0 +1,284 @@
+	}
+	rootsToTry := []string{absoluteRoot}
+
+	for _, customImport := range customImports {
+		absoluteCustomImport, err := filepath.Abs(customImport)
+		if err != nil {
+			return "", err
+		}
+		// Try the more specific custom imports first, rather than trying all of vendor
+		rootsToTry = append([]string{absoluteCustomImport}, rootsToTry...)
+	}
+
+	// Sort by length, so longer (more specific paths are attempted first)
+	sort.Slice(rootsToTry, func(i, j int) bool {
+		elementsJ := strings.Split(rootsToTry[j], string(os.PathSeparator))
+		elementsI := strings.Split(rootsToTry[i], string(os.PathSeparator))
+		return len(elementsI) > len(elementsJ)
+	})
+
+	var possibleImportPaths []string
+	for _, root := range rootsToTry {
+		if err := filepath.Walk(root, func(path string, info os.FileInfo, err error) error {
+			if strings.HasSuffix(path, importedProtoFile) {
+				importPath := strings.TrimSuffix(path, importedProtoFile)
+				possibleImportPaths = append(possibleImportPaths, importPath)
+
+			}
+			return nil
+		}); err != nil {
+			return "", err
+		}
+		// if found break
+		if len(possibleImportPaths) > 0 {
+			break
+		}
+	}
+	if len(possibleImportPaths) == 0 {
+		return "", errors.Errorf("found no possible import paths in root directory %v for import %v",
+			absoluteRoot, importedProtoFile)
+	}
+	if len(possibleImportPaths) != 1 {
+		log.Warnf("found more than one possible import path in root directory for "+
+			"import %v: %v",
+			importedProtoFile, possibleImportPaths)
+	}
+	return possibleImportPaths[0], nil
+
+}
+
+var defaultGogoArgs = []string{
+	"plugins=grpc",
+	"Mgoogle/protobuf/descriptor.proto=github.com/gogo/protobuf/protoc-gen-gogo/descriptor",
+	"Mgoogle/protobuf/any.proto=github.com/gogo/protobuf/types",
+	"Mgoogle/protobuf/wrappers.proto=github.com/gogo/protobuf/types",
+	"Mgoogle/protobuf/empty.proto=github.com/gogo/protobuf/types",
+	"Mgoogle/protobuf/struct.proto=github.com/gogo/protobuf/types",
+	"Mgoogle/protobuf/duration.proto=github.com/gogo/protobuf/types",
+	"Mgoogle/protobuf/timestamp.proto=github.com/gogo/protobuf/types",
+	"Menvoy/api/v2/discovery.proto=github.com/envoyproxy/go-control-plane/envoy/api/v2",
+	"gogoproto/gogo.proto=github.com/gogo/protobuf/gogoproto",
+}
+
+func (c *collector) writeDescriptors(protoFile, toFile string, imports []string, compileProtos bool) error {
+	cmd := exec.Command("protoc")
+	for i := range imports {
+		imports[i] = "-I" + imports[i]
+	}
+	cmd.Args = append(cmd.Args, imports...)
+	gogoArgs := append(defaultGogoArgs, c.customGogoArgs...)
+
+	if compileProtos {
+		cmd.Args = append(cmd.Args,
+			"--gogo_out="+strings.Join(gogoArgs, ",")+":"+c.descriptorOutDir,
+			"--ext_out="+strings.Join(gogoArgs, ",")+":"+c.descriptorOutDir,
+		)
+	}
+
+	cmd.Args = append(cmd.Args, "-o"+toFile, "--include_imports", "--include_source_info",
+		protoFile)
+
+	out, err := cmd.CombinedOutput()
+	if err != nil {
+		return errors.Wrapf(err, "%v failed: %s", cmd.Args, out)
+	}
+	return nil
+}
```

Comment:
how was this file changed from the original implementation? or was code simply moved? it's hard to tell without being able to compare with the original code side by side